src/utils/api/transaction/lsk.js

```diff
@@ -319,6 +321,17 @@ export const create = ({
 
   const schema = moduleAssetSchemas[moduleAssetId];
 
+  if (isHwSigning) {
+    // In case of hardware wallet, return signing bytes to HW device
+    const signingBytes = transactions.getSigningBytes(schema, transaction);
+    resolve({
+      networkIdentifier,
+      transactionObject: transaction,
+      transactionBytes: signingBytes,
+    });
+    return;
+  }
```

Comment:
This is wrong for a few reasons:
 - A function should either be a void or return a value in all conditions/branches. You're does not.
 - You are by passing the signature logic below this line and delegate the job to the React component which is an anti-pattern. Visual components must not handle business logic. They're harder to test and get affected by design changes when though the business logic remains unchanged.
 
 I suggest we break down this function, have 3 signature methods for
  - Regular signing
  - Multisignature signing 
  - Signing with HW
This will help create separation of concerns. as well as opening the door to any future signing flow.